test/unittest/StatisticsBackend/IsMetatrafficTests.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <database/entities.hpp>
+#include <StatisticsBackend.hpp>
+#include <StatisticsBackendData.hpp>
+
+using namespace eprosima::statistics_backend;
+using namespace eprosima::statistics_backend::database;
+
+constexpr const char* PID_DEFAULT = "36000";
+constexpr const char* DATA_TYPE_DEFAULT = "data_type";
+#define QOS_DEFAULT Qos(R"({"qos":"empty"})"_json)
+
+
+#define LOCATOR_DEFAULT_NAME(x) "locator_" + std::to_string(x)
+#define HOST_DEFAULT_NAME(x) "host_" + std::to_string(x)
+#define USER_DEFAULT_NAME(x) "user_" + std::to_string(x)
+#define PROCESS_DEFAULT_NAME(x) "process_" + std::to_string(x)
+#define DOMAIN_DEFAULT_NAME(x) "12" + std::to_string(x)
+#define ALIAS_DEFAULT_NAME(x) "domain_" + std::to_string(x)
+#define TOPIC_DEFAULT_NAME(x) "topic_" + std::to_string(x)
+#define PARTICIPANT_DEFAULT_NAME(x) "participant_" + std::to_string(x)
+#define DATAWRITER_DEFAULT_NAME(x) "datawriter_" + std::to_string(x)
+#define DATAREADER_DEFAULT_NAME(x) "datareader_" + std::to_string(x)
+
+#define GUID_DEFAULT(x) "01.0f.00.00.00.00.00.00.00.00.00.0" + std::to_string(x) + "|0.0.0.0"
+
+// Test metatraffic member of non-metatraffic topic
+TEST(is_metatraffic_tests, non_metatraffic_topic)
+{
+    // reset database
+    details::StatisticsBackendData::get_instance()->database_.reset(new Database);
+
+    std::shared_ptr<Domain> domain = std::make_shared<Domain>(std::string(DOMAIN_DEFAULT_NAME(0)));
+    domain->alias = ALIAS_DEFAULT_NAME(0);
+    details::StatisticsBackendData::get_instance()->database_->insert(domain);
+    // non-metatraffic topic
+    std::shared_ptr<Topic> topic = std::make_shared<Topic>(std::string(TOPIC_DEFAULT_NAME(
+                        0)), DATA_TYPE_DEFAULT, domain);
+    details::StatisticsBackendData::get_instance()->database_->insert(topic);
+    ASSERT_EQ(StatisticsBackend::is_metatraffic(topic->id), topic->metatraffic);
+    ASSERT_FALSE(StatisticsBackend::is_metatraffic(topic->id));
```

Comment:
Add a loop to check many topics, not just one by default.
Do the same for negative cases and for entities cases.
```suggestion
    for (auto topic_name : {<some_topic_names>}){
    std::shared_ptr<Topic> topic = std::make_shared<Topic>(std::string(topic_name(
                        0)), DATA_TYPE_DEFAULT, domain);
    details::StatisticsBackendData::get_instance()->database_->insert(topic);
    ASSERT_EQ(StatisticsBackend::is_metatraffic(topic->id), topic->metatraffic);
    ASSERT_FALSE(StatisticsBackend::is_metatraffic(topic->id));
    }
```